eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1

```diff
@@ -0,0 +1,60 @@
+<#yaml
+.SYNOPSIS
+Sync TypeSpec defintion and generate SDK for RPs under the input directory and have 'tsp-location.yaml'.
+
+.DESCRIPTION
+Sync TypeSpec defintion and generate SDK for RPs under the input directory and have 'tsp-location.yaml'.
+
+If the regenerated code is different from current code, this will tell the files the differences
+are in, and exit with a failure status.
+
+.PARAMETER Directory
+The directory that will be used to get 'tsp-location.yaml' and generate SDK. The default is the root directory of the
+Azure SDK for Java repository. One can also input service directory like: /sdk/storage, sdk/anomalydetector/azure-ai-anomalydetector.
+#>
+
+param(
+  [Parameter(Mandatory = $false)]
+  [string]$Directory
+)
+
+$path = ""
+if ($Directory) {
+  $path = $Directory
+}
```

Comment:
Do we need `$path` at all or can we just pass `$Directory` around?

Reply:
Thanks， removed $path